src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -821,8 +824,19 @@ leader_balancer_types::preference_index
 leader_balancer::build_preference_index() {
     leader_balancer_types::preference_index ret;
 
+    const bool should_sanction = _feature_table.should_sanction();
```

Comment:
I wonder why we can't just add this condition to `leadership_pinning_enabled()` and call it a day? Is it only to get nice warning messages? But the user is already getting license nag messages from `feature_manager`...

Reply:
> But the user is already getting license nag messages from feature_manager...

One of the design goals was to be as loud as possible with all the sanctionings, to minimise the possibility of someone trying to use an enterprise feature without a license, seeing that "it doesn't work" and thinking that it "just doesn't work" instead of understanding that this is an imposed restriction.

Reply:
@ztlpn we have 2 functions called `leadership_pinning_enabled`. One in "leader_balancer.cc" and one in "feature_manager.cc". And they are checking for different things. I guess that not checking for the preferences in "leader_balancer.cc" might be an optimisation, but shouldn't "feature_manager.cc" also check the  `_enable_rack_awareness` value?

Reply:
Semantics are a bit different, in feature_manager it is "has the user configured the feature", in leader_balancer it is "should redpanda do anything about it". `_enable_rack_awareness` here acts as a kill switch for all rack-related features, and the configuration when leadership pinning is configured, but _enable_rack_awareness is false doesn't make sense outside of incidents. A nice usability improvement would be to print some warning in this case, but it is out of scope here.

I propose not to complicate logic here and sanction and nag the user regardless of the value of `_enable_rack_awareness`.